saq/job.py

```diff
@@ -258,3 +268,77 @@ def replace(self, job):
         """Replace current attributes with job attributes."""
         for field in job.__dataclass_fields__:
             setattr(self, field, getattr(job, field))
+
+
+class PeriodicHeartbeat:
+    """Class to manage agent Heartbeats while Job is running"""
+
+    def __init__(self, job: Job) -> None:
```

Comment:
saq isn't using type annotations yet. I don't think they add much without using something like `mypy` in regular checks.

Can you remove this for now?
For documentation/IDE hints, use sphinx docstrings (as you do below).